HTCONDOR-1889 document htcondor eventlog (#1670)
htcondor/htcondor: docs/man-pages/htcondor.rst

| **htcondor** **eventlog** *follow* [ **-csv** | **-json**] [ **--groupby attribute** ] eventlog 

Description
-----------

*htcondor* is a tool for managing HTCondor jobs, job sets, resources, event logs, and
DAGs.  It can replace *condor_submit*, *condor_submit_dag*, *condor_q*,
*condor_status*, and *condor_userlog*, as well as all-new functionality and features.  The user interface is more consistent than its predecessor tools.

The first argument of the *htcondor* command (ignoring any global options) is
the *noun* representing an object in the HTCondor system to be operated on.
The nouns include an individual *job*, *jobset*, *eventlog*, or a *dag*.  Each
noun is then followed by a noun-specific *verb* that describe the operation on
that noun.

One of the following optional global option may appear before the noun:

Global Options
--------------

 **htcondor -h**, **htcondor --help**
     Display the help message. Can also be specified after any
     verb to display the options available for each verb.
 **htcondor -q ...**
     Reduce verbosity of log messages.
 **htcondor -v ...**
     Increase verbosity of log messages.

> Job Verbs

Reviewer: I liked the transition from the text of the man page to the block-formatted section for "global options," and kind of want to see it again for "job verbs", something like, "A noun-specific verb appears after each noun; the verbs are sorted by noun in the list, which includes with their individual option flags."

*htcondor*
===============

> Manage HTCondor jobs, job sets, dags, logs and resources

Reviewer: Oxford is disappoint.

Reviewer: Also, we should probably specify "event" logs, here.
```suggestion
Manage HTCondor jobs, job sets, dags, event logs, and resources
```

:index:`htcondor<single: htcondor; HTCondor commands>`\ :index:`htcondor command`

Synopsis
--------

**htcondor** [ **-h** | **--help** ] [ **-v** | **-q** ]

| **htcondor** **job** *submit* [**--resource** *resource-type*] [**--runtime** *time-seconds*] [**--email** *email-address*] submit_file
| **htcondor** **job** *status* [**--resource** *resource-type*] [**--skip-history**] job_id
| **htcondor** **job** *resources* [**--resource** *resource-type*] [**--skip-history**] job_id

| **htcondor** **jobset** *submit* description-file
| **htcondor** **jobset** *list* [**--allusers**]
| **htcondor** **jobset** *status* job-set-name [**--owner** *user-name*] [**--nobatch**] [**--skip-history**]
| **htcondor** **jobset** *remove* job-set-name [**--owner** *user-name*] [**--nobatch**] [**--skip-history**]

| **htcondor** **dag** *submit* dag-file
| **htcondor** **dag** *status* dagman-job-id

| **htcondor** **eventlog** *read* [ **-csv** | **-json**] [ **--groupby attribute** ] eventlog
| **htcondor** **eventlog** *follow* [ **-csv** | **-json**] [ **--groupby attribute** ] eventlog 

Description
-----------

> *htcondor* is a tool for managing HTCondor jobs, job sets, resources, logs and

Reviewer: ```suggestion
*htcondor* is a tool for managing HTCondor jobs, job sets, resources, event logs, and
```

> dags.  It can replace *condor_submit*, *condor_submit_dag*, *condor_q*,

Reviewer: ```suggestion
DAGs.  It can replace *condor_submit*, *condor_submit_dag*, *condor_q*,
```
The colloquialism is appropriate for the command-line; not sure about for the documentation.

> dags.  It can replace *condor_submit*, *condor_submit_dag*, *condor_q*,

Reviewer: Not
```suggestion
dags.  It has similar functionality to parts of *condor_submit*, *condor_submit_dag*, *condor_q*,
```
?

> *condor_status*, *condor_userlog*, and is intended to be a more self-consistent
> interface to the HTCondor system instead of those tools which have organically
> grown over decades.

Reviewer: ```suggestion
*condor_status*, and *condor_userlog*, as well as all-new functionality and features.  The user interface is more consistent than its predecessor tools.
```

The first argument of the *htcondor* command (ignoring any global options) is
the *noun* representing an object in the HTCondor system to be operated on.
The nouns include an individual *job*, *jobset*, *eventlog*, or a *dag*.  Each

> noun is then followed by a noun-specific verb that describe the operation on

Reviewer: ```suggestion
noun is then followed by a noun-specific *verb* that describe the operation on
```

that noun.

One of the following optional global option may appear before the noun:

Global Options
--------------

 **htcondor -h**, **htcondor --help**
     Display the help message. Can also be specified after any
     verb to display the options available for each verb.
 **htcondor -q ...**
     Reduce verbosity of log messages.
 **htcondor -v ...**
     Increase verbosity of log messages.

Job Verbs
---------

>  **htcondor job submit submit_file**

Reviewer: Not sure about the render here, but **submit_file** is not a literal here and this should be indicated; we do this above, in the summary, so perhaps:
```suggestion
 **htcondor job submit** *submit_file*
```

     Takes as an argument a submit file in the *condor_submit* job submit
     description language, and places a new job in an Access Point

>      **htcondor job submit Options**

Reviewer: ```suggestion
     Options
```
The indention in the rendered form makes it clear which options this applies to.  If you want to make it easier to search, I guess
```suggestion
     **htcondor job submit** Options
```
would be fine, but since this is actually a subheading, consider marking it as such.

>           **htcondor job submit --resource resource_type submit_file**

Reviewer: ```suggestion
          **htcondor job submit --resource** *resource_type submit_file*
```

>             Resource type used to run this job. Currently supports ``Slurm`` and ``EC2``.

Reviewer: Have you checked to see what this manpage looks like as a man page?

            Assumes the necessary setup is complete and security tokens available.
          **htcondor job submit --runtime runtime_in_seconds submit_file**
            Amount of time in seconds to allocate resources.
            Used in conjunction with the *--resource* flag.
          **htcondor job submit --email address submit_file**
            Email address to receive notification messages.
            Used in conjunction with the *--resource* flag.
    
    
 **htcondor job status**
     Takes as an argument a job id in the form of clusterid.procid,
     and returns a human readable presentation of the status
     of that job.

>      **job status option**

Reviewer: See previous suggestions.  Be consistent.  Don't use the lower-case "option" because it makes it look even more like a command-line constant.

>       **htcondor job status --skip-history job.id**

Reviewer: ```suggestion
      **htcondor job status --skip-history** *job.id*
```

        Passed to the *status* verb to skip checking history
        if job not found in the active job queue.

>  **htcondor job resources**

Reviewer: I'm OK with just dropping the arguments for these subheadings entirely, but be consistent.

     Takes as an argument a job id in the form of clusterid.procid,
     and returns a human readable presentation the machine resource
     used by this job.

Jobset Options
--------------

 **--allusers**
    Passed to the *list* verb to show job sets from all users
    rather than just the current user.
 **--nobatch**
    Passed to the *status* verb to display the status of
    individual job clusters within a job set
 **--owner=USERNAME**
    Passed to the *status* or *remove* verb to act on job sets
    submitted by the specified user instead of the current
    user. Using this option to *remove* job sets requires superuser
    permissions.
 **--skip-history**
    Passed to the *status* verb to skip checking history
    if job clusters are not found in the active job queue.

Eventlog Verbs
--------------

>  **htcondor eventlog read logfile**

Reviewer: ```suggestion
 **htcondor eventlog read** *logfile*
```

     Takes as an argument an event log to process.  It may be the per-job or

>      per-jobset eventlog, which was specified by the *log = some_file* in the

Reviewer: If "eventlog" is the preferred nomenclature, consider all my previous "event log"s corrected.

     submit description language.  For a dag, it can also be the *nodes.log*
     file that all dags generate.  Or, if the global event log is enabled by an

>      administrator with the *EVENT_LOG* configuration knob, it is the global

Reviewer: ```suggestion
     administrator with the *EVENT_LOG* configuration knob, it may be the global
```
